src/client/src/Domains/Messages/GetMessages/index.tsx

```diff
@@ -30,16 +30,21 @@ function Messages({ receiver }: MessagesProps) {
         if (receiver) sendRequest();
     }, [receiver, sendRequest]);
 
-    // Reload messaging component if new message is created between the
-    // logged-in user and the current receiver.
-    io.on('chat', (message: IMessage) => {
-        if (receiver &&
-            ((message.sender.id === receiver.id && message.receiver.id === user?.userId)
-                || (message.receiver.id === receiver.id && message.sender.id === user?.userId))) {
-            sendRequest();
-            // messages.push(message);
-        }
-    });
+    React.useEffect(() => {
+        // Update io listener by every receiver update.
+        io.removeListener('message_area');
+        io.on('message_area', (message: IMessage) => {
+            // Reload messaging component if new message is created between the
```

Comment:
wait, so it keeps doing a reload even though the request is not being sent?

Reply:
What do you mean?

Reply:
It reloads when a new message is created between the logged-in user and the current receiver.

Reply:
Or, it appends a new message to the "messages" list if the new message is sent by the logged-in user or the currently selected receiver.

Reply:
oh so, the feature doesn't look like let's say FB messenger chat so far. right? Users have to fill out the NewMessageForm to create a message

Reply:
oh okay so it's not reloading the page then, it's just appending the message to the list of messages

Reply:
Done.